### PR TITLE
HIVE-28130: Iceberg: Add support for Rename Branch.

### PR DESCRIPTION
**Pull request by ayushtkn**
### What changes were proposed in this pull request?

Add support for renaming an iceberg branch

### Why are the changes needed?

Better Usability of the feature

### Does this PR introduce _any_ user-facing change?

Iceberg Branches can be renamed

### Is the change a dependency upgrade?

No

### How was this patch tested?

UT

### CODE REVIEW

File: parser/src/java/org/apache/hadoop/hive/ql/parse/AlterClauseParser.g [hunk 1]
```diff
@@ -503,6 +503,8 @@ alterStatementSuffixExecute
     -> ^(TOK_ALTERTABLE_EXECUTE KW_EXPIRE_SNAPSHOTS KW_RETAIN $numToRetain)
     | KW_EXECUTE KW_DELETE KW_ORPHAN_FILES (KW_OLDER KW_THAN LPAREN (timestamp=StringLiteral) RPAREN)?
     -> ^(TOK_ALTERTABLE_EXECUTE KW_ORPHAN_FILES $timestamp?)
+    | KW_EXECUTE KW_RENAME KW_BRANCH sourceBranch=StringLiteral KW_TO targetBranch=StringLiteral
```
Comment (deniskuzZ) [inline parser/src/java/org/apache/hadoop/hive/ql/parse/AlterClauseParser.g:506]: why do we need `execute` here? simple rename should be enough
````
alter table ice01 rename branch 'source' to 'target';
````

Comment (zhangbutao) [reply parser/src/java/org/apache/hadoop/hive/ql/parse/AlterClauseParser.g:506]: Agreed with @deniskuzZ .  
I think we need to reach an agreement on when to use this `execute` syntax.
IMHO, `execute` is good for some optimization & maintenance operation. I think TrinoDB also thought so:(https://trino.io/docs/current/connector/iceberg.html#alter-table-execute)

For basic operation, rename can use the same simple syntax like create/drop.

Comment (ayushtkn) [reply parser/src/java/org/apache/hadoop/hive/ql/parse/AlterClauseParser.g]: thanx folks, I have changed the syntax & pushed

File: ql/src/java/org/apache/hadoop/hive/ql/ddl/table/snapshotref/branch/rename/AlterTableRenameBranchAnalyzer.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import org.apache.hadoop.hive.ql.parse.AlterTableSnapshotRefSpec;
+import org.apache.hadoop.hive.ql.parse.HiveParser;
+import org.apache.hadoop.hive.ql.parse.SemanticException;
+
+@DDLSemanticAnalyzerFactory.DDLType(types = HiveParser.TOK_ALTERTABLE_RENAME_BRANCH)
+public class AlterTableRenameBranchAnalyzer extends AbstractAlterTableAnalyzer {
+
+  protected AlterTableType alterTableType;
+
+  public AlterTableRenameBranchAnalyzer(QueryState queryState) throws SemanticException {
```
Comment (zhangbutao) [inline ql/src/java/org/apache/hadoop/hive/ql/ddl/table/snapshotref/branch/rename/AlterTableRenameBranchAnalyzer.java:44]: Nit: Maybe use the name `AlterTableRenameSnapshotRefAnalyzer`? Just like `AlterTableCreateSnapshotRefAnalyzer`.

But i know iceberg tag can not be renamed, so this class can only be used for branch. I am not sure if we need to follow the previous naming rules.  cc @deniskuzZ

File: ql/src/java/org/apache/hadoop/hive/ql/parse/AlterTableSnapshotRefSpec.java [hunk 3]
```diff
@@ -122,4 +122,29 @@ public String toString() {
       return MoreObjects.toStringHelper(this).add("refName", refName).add("ifExists", ifExists).toString();
     }
   }
+
+  public static class RenameBranchSpec {
```
Comment (zhangbutao) [inline ql/src/java/org/apache/hadoop/hive/ql/parse/AlterTableSnapshotRefSpec.java:126]: Also here, should we use `RenameSnapshotrefSpec`?

**Review by zhangbutao — COMMENTED** [review]
Just left two nit comments, others looks good to me. Thanks.

File: ql/src/java/org/apache/hadoop/hive/ql/ddl/table/snapshotref/branch/rename/AlterTableRenameBranchAnalyzer.java (hunk 2 above)
Comment (ayushtkn) [reply ql/src/java/org/apache/hadoop/hive/ql/ddl/table/snapshotref/branch/rename/AlterTableRenameBranchAnalyzer.java]: I changed it, who knows those guys tmrw start supporting it for tags as well :-)

File: ql/src/java/org/apache/hadoop/hive/ql/parse/AlterTableSnapshotRefSpec.java (hunk 3 above)
Comment (ayushtkn) [reply ql/src/java/org/apache/hadoop/hive/ql/parse/AlterTableSnapshotRefSpec.java]: Done

**Review by zhangbutao — APPROVED** [review]
LGTM + 1 Pending a green CI.

**Pull request merged**